policyengine_api/utils/logger.py

```diff
@@ -0,0 +1,116 @@
+        self.id = (
+            id if id is not None else datetime.now().strftime("%Y%m%d_%H%M%S")
+        )
+
+        self.logger = logging.getLogger(self.name)
+        self.logger.setLevel(logging.INFO)
+
+        # Create log directory if it doesn't exist
+        try:
+            self.dir.mkdir(parents=True, exist_ok=True)
+        except Exception as e:
+            print(
+                f"Warning: Could not create log directory {self.dir}: {str(e)}"
+            )
+            # Fall back to current directory
+            self.dir = Path(".")
+
+        # Create log file path based upon directory
+        self.filepath = self.dir.joinpath(f"{self.name}_{self.id}.log")
+
+        self.cloud_client: cloud_logging.Client = None
+
+        # Prevent duplicate handlers
+        if not self.logger.handlers:
+            # File handler - logs to local file
+            file_handler = logging.FileHandler(str(self.filepath))
+            file_handler.setLevel(logging.INFO)
+            file_handler.setFormatter(
+                logging.Formatter(
+                    "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+                )
+            )
+            self.logger.addHandler(file_handler)
+
+            # Console handler
+            console_handler = logging.StreamHandler(sys.stdout)
+            console_handler.setLevel(logging.INFO)
+            console_handler.setFormatter(
+                logging.Formatter(
+                    "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+                )
+            )
+            self.logger.addHandler(console_handler)
+
+            # Google Cloud Logging handler; don't log to GCP if in debug
+            if log_to_cloud and os.environ.get("FLASK_DEBUG") != "1":
+                try:
+                    cloud_client = cloud_logging.Client()
+                except Exception as e:
+                    print(f"Google Cloud Logging error: {str(e)}")
+                    return
+                cloud_handler = cloud_logging.handlers.CloudLoggingHandler(
+                    cloud_client, name=f"{self.name}"
+                )
+                cloud_handler = cloud_logging.handlers.CloudLoggingHandler(
+                    cloud_client
+                )
+                cloud_handler.setLevel(logging.INFO)
+                cloud_handler.setFormatter(
+                    logging.Formatter("%(levelname)s - %(message)s")
+                )
+                self.logger.addHandler(cloud_handler)
+
+    def log(self, message: str, level: str = "info", **context):
+        """
+        Log a message with optional context data
+
+        Args:
+            message (str): Message to log
+            level (str): Log level (default: "info")
+            context (dict): Optional context data to log
+        """
+
+        # Format message with context if provided
+        if context:
+            context_str = " ".join(f"{k}={v}" for k, v in context.items())
```

Comment:
This makes more sense to me in terms of standardizing your message string. you don't necessarily need to wrap logger to do this though, you could just write a 'format_context' function or whatever:

```
logger.info(format_context(message, context))
```